src/legacy/server/saved_objects/import/validate_references.test.ts

```diff
@@ -0,0 +1,268 @@
+            "id": "5",
+            "type": "index-pattern",
+          },
+          Object {
+            "id": "6",
+            "type": "index-pattern",
+          },
+          Object {
+            "id": "7",
+            "type": "search",
+          },
+        ],
+        "type": "missing_references",
+      },
+      "id": "4",
+      "type": "visualization",
+    },
+  ],
+  "filteredObjects": Array [
+    Object {
+      "attributes": Object {},
+      "id": "1",
+      "references": Array [],
+      "type": "visualization",
+    },
+  ],
+}
+`);
+    expect(savedObjectsClient.find).toHaveBeenCalledTimes(1);
+  });
+
+  test(`doesn't return errors when references exist in Elasticsearch`, async () => {
+    savedObjectsClient.find.mockResolvedValue({
+      saved_objects: [
+        {
+          id: '1',
+          type: 'index-pattern',
+          attributes: {},
+          references: [],
+        },
+      ],
+    });
+    const savedObjects = [
+      {
+        id: '2',
+        type: 'visualization',
+        attributes: {},
+        references: [
+          {
+            name: 'ref_0',
+            type: 'index-pattern',
+            id: '1',
+          },
+        ],
+      },
+    ];
+    const result = await validateReferences(savedObjects, savedObjectsClient);
+    expect(result).toMatchInlineSnapshot(`
+Object {
+  "errors": Array [],
+  "filteredObjects": Array [
+    Object {
+      "attributes": Object {},
+      "id": "2",
+      "references": Array [
+        Object {
+          "id": "1",
+          "name": "ref_0",
+          "type": "index-pattern",
+        },
+      ],
+      "type": "visualization",
+    },
+  ],
+}
+`);
+    expect(savedObjectsClient.find).toHaveBeenCalledTimes(1);
+  });
+
+  test(`doesn't return errors when references exist within the saved objects`, async () => {
+    savedObjectsClient.find.mockResolvedValue({
+      saved_objects: [],
+    });
+    const savedObjects = [
+      {
+        id: '1',
+        type: 'index-pattern',
+        attributes: {},
+        references: [],
+      },
+      {
+        id: '2',
+        type: 'visualization',
+        attributes: {},
+        references: [
+          {
+            name: 'ref_0',
+            type: 'index-pattern',
+            id: '1',
+          },
+        ],
+      },
+    ];
+    const result = await validateReferences(savedObjects, savedObjectsClient);
+    expect(result).toMatchInlineSnapshot(`
+Object {
+  "errors": Array [],
+  "filteredObjects": Array [
+    Object {
+      "attributes": Object {},
+      "id": "1",
+      "references": Array [],
+      "type": "index-pattern",
+    },
+    Object {
+      "attributes": Object {},
+      "id": "2",
+      "references": Array [
+        Object {
+          "id": "1",
+          "name": "ref_0",
+          "type": "index-pattern",
+        },
+      ],
+      "type": "visualization",
+    },
+  ],
+}
+`);
+    expect(savedObjectsClient.find).toHaveBeenCalledTimes(1);
+  });
+});
```

Comment:
I think we could use a test or two to make sure that only `ENFORCED_TYPES` are checked.